fenix/app/src/main/java/org/mozilla/fenix/addons/ExtensionsProcessDisabledForegroundController.kt

```diff
@@ -20,35 +18,30 @@ import mozilla.components.support.webextensions.ExtensionsProcessDisabledPromptO
 import org.mozilla.fenix.R
 import org.mozilla.fenix.components.AppStore
 import org.mozilla.fenix.ext.components
-import kotlin.system.exitProcess
 
 /**
  * Controller for handling extensions process spawning disabled events. When the app is in
  * foreground this will call for a dialog to decide on correct action to take (retry enabling
- * process spawning or disable extensions). When in background, we kill the app to prevent
- * extensions from being disabled and network requests continuing.
+ * process spawning or disable extensions).
  *
  * @param context to show the AlertDialog
  * @param browserStore The [BrowserStore] which holds the state for showing the dialog
  * @param appStore The [AppStore] containing the application state
  * @param builder to use for creating the dialog which can be styled as needed
  * @param appName to be added to the message. Optional and mainly relevant for testing
- * @param onKillApp called when the app is backgrounded and extensions process is disabled
  */
-class ExtensionsProcessDisabledController(
+class ExtensionsProcessDisabledForegroundController(
     @UiContext context: Context,
```

Comment:
Thanks for putting the patch!
In general the solution looks good, but could we address the issue by just using a `WeakReference` of `Context`?

Reply:
Thanks @Amejia481 ! In theory yes, I didn't try that and test it though.

Reply:
I really like your suggestion 💪🏽 , but it will be nice if we could give a try, as it maybe a bit simpler and it's pattern that more widely use in the code base :)

Reply:
Sounds reasonable! I also like the solution since it separates two different actions we take based on foreground or background, but I also went with the aim of keeping changes to the minimum, I'll give that a shot!